### ISSUE TITLE
Support variable such as `CURRENT_YEAR` used by friendly-snippets

### ISSUE THREAD

**Issue by jmetz** [estin/simple-completion-language-server#68]
There are a bunch of common system variables that spring to mind, including date-time related, username, email, or perhaps even the result of a shell command.

**Comment by jmetz** [comment]
Yeah that looks pretty comparable to other snippet systems I've used before 👍

**Comment by GameDungeon** [comment]
Those would be great, especially TM_SELECTED_TEXT which allows for way more powerful snippets.

**Comment by GameDungeon** [comment]
I just checked the code for helix-editor/helix#9801 (which is the branch I use). It currently implements variable parsing, but it does not seem to have support for any specific variables yet, but the fact it has got that far is good news.
